std/permutation.asm

```diff
@@ -0,0 +1,39 @@
+use std::prover::challenge;
+use std::array::fold;
+use std::array::len;
+use std::convert::int;
+use std::check::assert;
+
+let is_first: col = |i| if i == 0 { 1 } else { 0 };
+
+let alpha: expr = challenge(0, 1);
```

Comment:
We should add a comment that this conflicts with any other mechanism that uses challenges.

Reply:
I added a comment. I wonder if this is a problem though! For example, rn, the same two challenges would be used for all permutations. But is this a problem? I think not, because the arguments don't interact in any way. Would it be a problem if the challenge was used in any other way that's independent of the permutation?